lib/scoped_search/definition.rb

```diff
@@ -21,41 +21,71 @@ class Field
 
       # Initializes a Field instance given the definition passed to the
       # scoped_search call on the ActiveRecord-based model class.
-      def initialize(definition, options = {})
-        @definition = definition
-        @definition.profile = options[:profile] if options[:profile]
-        @definition.default_order ||= default_order(options)
-
-        case options
-        when Symbol, String
-          @field = field.to_sym
-        when Hash
-          @field = options.delete(:on)
-
-          # Set attributes from options hash
-          @complete_value   = options[:complete_value]
-          @relation         = options[:in]
-          @key_relation     = options[:in_key]
-          @key_field        = options[:on_key]
-          @offset           = options[:offset]
-          @word_size        = options[:word_size] || 1
-          @ext_method       = options[:ext_method]
-          @operators        = options[:operators]
-          @only_explicit    = !!options[:only_explicit]
-          @full_text_search = options[:full_text_search]
-          @default_operator = options[:default_operator] if options.has_key?(:default_operator)
-          @complete_enabled = options[:complete_enabled].nil? ? true : options[:complete_enabled]
-          @validator        = options[:validator]
+      #
+      # Field name may be given in positional 'field' argument or 'on' named
+      # argument.
+      def initialize(definition,
+                     field = nil,
+                     aliases: [],
+                     complete_enabled: true,
+                     complete_value: nil,
+                     default_operator: nil,
+                     default_order: nil,
+                     ext_method: nil,
+                     full_text_search: nil,
+                     in_key: nil,
+                     offset: nil,
+                     on: field,
+                     on_key: nil,
+                     only_explicit: nil,
+                     operators: nil,
+                     profile: nil,
+                     relation: nil,
+                     rename: nil,
+                     validator: nil,
+                     word_size: 1,
+                     **kwargs)
```

Comment:
This is on line 62:

    raise ArgumentError, "Unknown arguments to scoped_search: #{kwargs.keys.join(', ')}" unless kwargs.empty?

Actually _this_ is the workaround for Ruby 2.0. If we dropped 2.0 support then I could use `in` and `alias` in the argument list, remove `**kwargs`, and rely on Ruby's built-in error message. Using 2.1+ we can write `binding.local_variable_get(:in)` instead of `kwargs[:in]`.